contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -0,0 +1,138 @@
+use soroban_sdk::{Address, BytesN, Env, Vec};
+use stellar_axelar_std::interfaces::{CustomMigratableInterface, UpgradableClient};
+
+use crate::error::ContractError;
+use crate::{deployer, storage, InterchainTokenService};
+
+pub mod legacy_storage {
+    use soroban_sdk::{contracttype, Address, BytesN, String};
+    use stellar_axelar_std::contractstorage;
+
+    use crate::storage::TokenIdConfigValue;
+
+    #[contractstorage]
+    enum DataKey {
+        #[instance]
+        #[value(Address)]
+        Gateway,
+
+        #[instance]
+        #[value(Address)]
+        GasService,
+
+        #[instance]
+        #[value(String)]
+        ChainName,
+
+        #[instance]
+        #[value(String)]
+        ItsHubAddress,
+
+        #[instance]
+        #[value(Address)]
+        NativeTokenAddress,
+
+        #[instance]
+        #[value(BytesN<32>)]
+        InterchainTokenWasmHash,
+
+        #[instance]
+        #[value(BytesN<32>)]
+        TokenManagerWasmHash,
+
+        #[persistent]
+        #[status]
+        TrustedChain { chain: String },
+
+        #[persistent]
+        #[value(TokenIdConfigValue)]
+        TokenIdConfig { token_id: BytesN<32> },
+
+        #[persistent]
+        #[value(i128)]
+        FlowLimit { token_id: BytesN<32> },
+
+        #[temporary]
+        #[value(i128)]
+        FlowOut { flow_key: FlowKey },
+
+        #[temporary]
+        #[value(i128)]
+        FlowIn { flow_key: FlowKey },
+    }
+
+    #[contracttype]
+    #[derive(Clone, Debug)]
+    pub struct FlowKey {
+        pub token_id: BytesN<32>,
+        pub epoch: u64,
+    }
+}
+
+impl CustomMigratableInterface for InterchainTokenService {
+    type MigrationData = (
+        BytesN<32>, /* new_wasm_hash */
+        Vec<(BytesN<32> /* token_id */, u64 /* epoch */)>,
```

Comment:
only the current epoch needs to be migrated. The old epochs get automatically deleted due to the use of temporary storage, and are never accessed once the epoch changes